tools/customization_dataset_preparation/customization_dataset_preparation.py

```diff
@@ -79,31 +80,50 @@ def load_file_into_df(filename):
     return df, message
 
 
+def recommend_hyperparameters_human_readable(recommended_hyperparameters):
+    message = 'TODO: Recommended hyperparameters\n'
+    for param, param_value in recommended_hyperparameters.items():
+        message += f'{param}: {param_value}\n'
+    return message
+
+
 def recommend_hyperparameters(df, model=None):
     """
     Makes recommendations on the batch_size to use for training, based on the dataset size
+    
+    All hyperparameters except batch_size and max_batch_size are hardcoded based on API defaults for now
     """
     potential_batch_sizes = [2, 4, 8, 12, 16, 32, 64, 128]
     bs = 2
     for potential_bs in potential_batch_sizes:
         if 0.002 * len(df) > potential_bs:
             bs = potential_bs
 
-    message = f"TODO: A batch_size={bs} is recommended for training."
-
+    max_bs = 128
     if len(df) < 128:
         max_bs = 2
         for potential_bs in potential_batch_sizes:
             if potential_bs < len(df) * 0.9:
                 max_bs = potential_bs
-        additional_msg_for_small_df = f" Please have a maximum batch_size of {max_bs}."
-        message += additional_msg_for_small_df
-
-    return message
 
-
-def estimating_customization_job_time(df, recommend_hyperparameters_message):
-    recommended_batch_size = int(recommend_hyperparameters_message.split("=")[1].split()[0])
+    df_char_length = df.apply(lambda x: len(x.prompt) + len(x.completion), axis=1)
+    length_by_chars = sorted(list(df_char_length))
+    n_samples_under_99p5_limit = math.ceil(len(df_char_length) * 0.995)
+    max_char_length = length_by_chars[n_samples_under_99p5_limit - 1]
+    # every token is around 4 chars + 100 for extra capacity
+    max_seq_length = max_char_length // 4 + 100
```

Comment:
What if the users data is very uniform in terms of seq length? E.g. the 99p5_limit is off by very small number from average? 
I would only use 99p5_limit if it differs significantly from the mean + 2*std. E.g. - we should be cutting outliers and if there is no outliers we shouldn't be cutting.

Reply:
Hmm I think this current logic might work because if say we have 1000 examples with range([1. 1000]) chars each. Then the max limit will be 995 chars. When calculating the max_seq_length, we include an additional 100 tokens (partly for this and partly for the fact that calculation based on chars/tokens is inexact) so we can approximately consider ~400 additional characters. What this means to say is in the general case, we would likely not cut off any samples unless it has >400 characters more than the 99p5 limit. Do you think this will be enough? 

If it's not, I can also calculate mean + 2*std and use that to find max_char_length but that code will be a bit hard to read (since we need to first do 

`max_char_length = max( min(mean+2*std , length_by_chars[-1]), length_by_chars[n_samples_under_99p5_limit - 1]) `

and then add some spare tokens to account imprecise char/token).